### PR TITLE
Propagate ImportChain to synthesized methods

### PR DESCRIPTION
**Pull request by cston**
Imports were not included in MoveNext method generated for async lambdas.

### CODE REVIEW

**Comment by cston** [comment]
@gafter, @AlekseyTs, @dotnet/interactive

File: src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs [hunk 1]
```diff
@@ -579,69 +579,81 @@ private void CompileSynthesizedMethods(TypeCompilationState compilationState)
...
 
-            foreach (var methodWithBody in compilationState.SynthesizedMethods)
+            var oldImportChain = compilationState.CurrentImportChain;
+            try
             {
-                var method = methodWithBody.Method;
+                foreach (var methodWithBody in synthesizedMethods)
+                {
+                    var importChain = methodWithBody.ImportChainOpt;
+                    compilationState.CurrentImportChain = importChain;
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs]: Subtle.  I didn't realize this was being consumed.

**Comment by amcasey** [comment]
:+1:

**Comment by MattGertz** [comment]
Approved pending successful test completion.

**Pull request merged**